src/features/hotspots/setup/NotHotspotOwnerError.tsx

```diff
@@ -1,44 +1,182 @@
+        Animated.timing(slideUpAnimRef.current, {
+          toValue: -8,
+          duration: 700,
+          useNativeDriver: true,
+          easing: Easing.bounce,
+        }),
+        Animated.timing(slideUpAnimRef.current, {
+          toValue: 0,
+          duration: 700,
+          useNativeDriver: true,
+        }),
+      ]),
+      { iterations: 3 },
+    ).start(() => setAnimFinished(true))
+  }, [following])
+
+  useEffect(() => {
+    if (!animFinished) return
+
+    Animated.timing(opacityAnim.current, {
+      toValue: 0,
+      duration: 500,
+      easing: Easing.linear,
+      useNativeDriver: true,
+    }).start()
+  }, [animFinished])
+
+  useEffect(() => {
+    anim()
+  }, [anim, following])
+
+  useEffect(() => {
+    return navigation.addListener('focus', () => {
+      dispatch(fetchHotspotsData())
+    })
+  }, [navigation, dispatch])
+
+  const handleFollowChange = useCallback((nextFollowing) => {
+    setFollowing(nextFollowing)
+  }, [])
+
+  const colors = useMemo(
+    () => ({ following: followPurple, notFollowing: '#40466F' }),
+    [followPurple],
+  )
 
   const navExit = useCallback(async () => {
     navigation.navigate('MainTabs')
   }, [navigation])
 
+  const hotspotNameStyle = useMemo(
+    () => ({ color: following ? white : '#585D89' }),
+    [following, white],
+  )
+
   return (
     <BackScreen>
       <Box flex={1} justifyContent="center" paddingBottom="xxl">
-        <Text variant="h1" marginBottom="l" maxFontSizeMultiplier={1}>
+        <Text
+          variant="h1s"
+          marginBottom="lm"
+          numberOfLines={2}
+          adjustsFontSizeToFit
+          maxFontSizeMultiplier={1}
+        >
           {t('hotspot_setup.not_owner.title')}
         </Text>
         <Text
           variant="subtitleMedium"
           color="purpleMain"
-          marginBottom={{ phone: 'l', smallPhone: 'ms' }}
+          numberOfLines={2}
+          maxFontSizeMultiplier={1.1}
+          adjustsFontSizeToFit
+          marginBottom={{ phone: 'xl', smallPhone: 'ms' }}
         >
           {t('hotspot_setup.not_owner.subtitle_1')}
         </Text>
         <Text
           variant="subtitleLight"
-          marginBottom={{ phone: 'xl', smallPhone: 'ms' }}
-          numberOfLines={2}
-          adjustsFontSizeToFit
+          color="white"
+          marginBottom={{ phone: 'l', smallPhone: 'ms' }}
+          fontSize={15}
+          lineHeight={20}
           maxFontSizeMultiplier={1.3}
         >
           {t('hotspot_setup.not_owner.subtitle_2')}
         </Text>
+        <Box
+          height={52.5}
+          style={styles.ownedHotspotBox}
+          borderRadius="l"
+          alignItems="center"
+          flexDirection="row"
+        >
+          <Text
+            marginLeft="l"
+            style={hotspotNameStyle}
+            variant="medium"
+            fontSize={16}
+            flex={1}
+          >
+            {startCase(name)}
+          </Text>
+          <FollowButton
```

Comment:
just wondering if it's obvious enough that only the flag here is pressable? it sort of looks like the whole box is a button. Though I know we're trying to keep all the follow logic contained within the `FollowButton` component which is great btw. I guess with the animation maybe that points out the flag enough. we can see what happens with users

Reply:
yeah, we could always add children to the FollowButton, but let's wait and get feedback.